packages/website/docs/using-react.md

```diff
@@ -0,0 +1,127 @@
+---
+id: using-react
+title: Using Autocomplete with React
+---
+
+Learn how to create and use an React Autocomplete component.
+
+This guide shows how to create a React Autocomplete component. It uses the [`useRef`](https://reactjs.org/docs/hooks-reference.html#useref) and [`useEffect`](https://reactjs.org/docs/hooks-reference.html#useeffect) hooks to create and mount the component. It doesn't define specific [sources](sources). Rather, you can pass [sources](sources) and other [options](basic-options) as [props](https://reactjs.org/docs/components-and-props.html).
+
+## Prerequisites
+
+This tutorial assumes that you have:
+- an existing [React (v 16.8+) application](https://reactjs.org/docs/getting-started.html) where you want to implement the autocomplete dropdown
+- familiarity with the [basic Autocomplete configuration options](basic-options)
+
+## Creating the component
+
+Start with some boilerplate for creating a React component. This component uses the [`useRef`](https://reactjs.org/docs/hooks-reference.html#useref) hook to create a mutable ref object, `containerRef`, to mount the autocomplete on. To learn more about this hook, check out the [`useRef` React documentation](https://reactjs.org/docs/hooks-reference.html#useref).
+
+ All that you need to render is a `div` with the `containerRef` as the `ref`.
+
+```jsx title="Autocomplete.jsx"
+import React, { createElement, Fragment, useEffect, useRef } from 'react';
+import { render } from 'react-dom';
+
+export function Autocomplete(props) {
+  const containerRef = useRef(null);
+
+  useEffect(() => {
+    if (!containerRef.current) {
+      return undefined;
+    }
+  }, [props]);
+
+  return (
+    <div
+      ref={containerRef}
+    />
+  );
+}
+```
+
+### Mounting the autocomplete
+
+Now that you have access to the DOM through the `containerRef` object, you can create and mount the autocomplete instance.
+
+You can use the same [Autocomplete options](basic-options), but you must also pass the `renderer` and `render` parameters. You can rely on `props` to pass any autocomplete options you want to remain configurable.
+
+Don't forget to [clean up the effect](https://reactjs.org/docs/hooks-reference.html#cleaning-up-an-effect) by returning a function that destroys the autocomplete instance.
+
+```jsx title="Autocomplete.jsx"
+import { autocomplete } from '@algolia/autocomplete-js';
+import React, { createElement, Fragment, useEffect, useRef } from 'react';
+import { render } from 'react-dom';
+
+export function Autocomplete(props) {
+  const containerRef = useRef(null);
+
+  useEffect(() => {
+    if (!containerRef.current) {
+      return undefined;
+    }
+
+    const search = autocomplete({
+      container: containerRef.current,
+      placeholder: 'Search',
```

Comment:
We can remove this option to be on par with the default Autocomplete JS options.

```suggestion
```